src/Microsoft.AspNet.Mvc.Core/Description/DefaultApiDescriptionProvider.cs

```diff
@@ -450,5 +385,279 @@ private IApiResponseMetadataProvider[] GetResponseMetadataAttributes(ControllerA
+                    // doesn't use any IBinderMetadata.
+                    // 
+                    // Add a single 'default' parameter description for the model.
+                    Context.Results.Add(CreateResult(modelMetadata, source, containerName: string.Empty));
+                }
+            }
+
+            private bool Visit(ModelMetadata modelMetadata, ApiParameterSource ambientSource, string containerName)
+            {
+                ApiParameterSource source;
+                if (TryGetSource(modelMetadata, out source))
+                {
+                    // We have a definite answer for this model. This is a greedy source like
+                    // [FromBody] so there's no need to consider properties.
+                    Context.Results.Add(CreateResult(modelMetadata, source, containerName));
+
+                    return true;
+                }
+             
+                if (!modelMetadata.Properties.Any())
+                {
+                    if (source == null)
+                    {
+                        // If there are no properties and no source then we don't know how to bind this.
+                        // Return and let it get potentially merged with the container.
+                        return false;
+                    }
+                    else
+                    {
+                        // We found a new source, and this model has no properties. This is probabaly
+                        // a simple type with an attribute like [FromQuery].
+                        Context.Results.Add(CreateResult(modelMetadata, source, containerName));
+                        return true;
+                    }
+                }
+
+                // This will come from composite model binding - so investigate what's going on with each property.
+                // 
+                // Basically once we find something that we know how to bind, we want to treat all properties at that
+                // level (and higher levels) as separate parameters. 
+                //
+                // Ex:
+                //
+                //      public IActionResult PlaceOrder(OrderDTO order) {...}
+                //
+                //      public class OrderDTO
+                //      {
+                //          public int AccountId { get; set; }
+                //          
+                //          [FromBody]
+                //          public Order { get; set; }
+                //      }
+                //
+                // This should result in two parameters:
+                //
+                //  AccountId - source: Any
+                //  Order - source: Body
+                //
+
+                var propertyCount = 0;
+                var unboundProperties = new HashSet<ModelMetadata>();
+
+                // We don't want to append the parameter name to the model name when recursing.
```

Comment:
Is this correct? wouldnt the containerType be non null when visiting inner properties?